rpcperms/interceptor.go

```diff
@@ -929,6 +929,14 @@ func (r *InterceptorChain) acceptRequest(requestID uint64,
 			continue
 		}
 
+		var err error
+		msg.CustomCaveatCondition, err = macaroons.GetCustomCaveatCondition(msg.Macaroon, middleware.customCaveatName)
```

Comment:
We'll probably want the same in the `interceptResponse` call below.

Reply:
Ah, and we should just ignore the error here. Otherwise in read-only mode this will fail if a normal macaroon without custom condition comes in.

Reply:
I was unsure about whether this logic was needed in `interceptResponse` as well - it's still not clear to me which is used when but I'll add this logic there for good measure.

And good catch about ignoring the error, I overlooked the times when we'll be read only but not have a custom caveat and get past the `if` check above. if think I'll just wrap this in a `if hasCustomCaveat` statement

Reply:
Ignoring the error felt a bit cleaner on second thought, less boilerplate code and we just get back an empty string anyway.